test/strategies.test.js

```diff
@@ -63,6 +63,100 @@ describe('RetryStrategies', function () {
         t.strictEqual(500, response.statusCode);
         done();
       });
+    });
+
+    it('should allow boolean true return value', function (done) {
+      var strategy = function (err, response, body) {
+        return true;
+      };
+
+      request({
+        url: 'http://www.filltext.com/?rows=1',
+        maxAttempts: 2,
+        retryDelay: 100,
+        retryStrategy: strategy
+      }, function(err, response, body) {
+        if(err) done(err);
+
+        t.strictEqual(2, this.attempts); // maxAttempts reached
+        done();
+      });
+    });
+
+    it('should allow boolean false return value', function (done) {
+      var strategy = function (err, response, body) {
+        return false;
+      };
+
+      request({
+        url: 'http://www.filltext.com/?rows=1',
+        maxAttempts: 2,
+        retryDelay: 100,
+        retryStrategy: strategy
+      }, function(err, response, body) {
+        if(err) done(err);
+
+        t.strictEqual(1, this.attempts); // maxAttempts NOT reached
+        done();
+      });
+    });
+
+    it('should allow Error object as true return value', function (done) {
+      var strategy = function (err, response, body) {
+        return Error('error');
```

Comment:
if the retryStrategy yield an Error object should be instead call the `callback(err)` directly ?
Accepted return values for retryStrategy can only be true/false from my point of view.

Reply:
While the docs do state the accepted return value as boolean, the example is returning `err || response.statusCode === 502`. We had to force similar code to a boolean return value to workaround issue https://github.com/FGRibreau/node-request-retry/issues/98 that popped up when updating our dependencies. However, that will be unnecessary once this fix is in place, and backwards compatibility will be restored.

Shall I:

- update the example to force a boolean return value and match the docs
- test for `Error` and `statusCode` values for backwards compatibility with the example
- just remove this test as it is over-complicating things, and maybe change my commit message

Reply:
ping @FGRibreau

Reply:
Yes, please update the example to force a boolean return value and match the docs and remove this test :)

Reply:
OK, I've updated the example, removed the test, and updated the commit message. I hope a force-push is ok.